back/queries/adventure.js

```diff
@@ -68,9 +68,30 @@ async function RateQuery(reviews, username, experienceId) {
   return hasError;
 }
 
+// Update preferences query
+// Used by routes/adventure/preferences/update
+async function UpdatePreferenceQuery(username, prioritize, activeIDs, minValues, hasMinValues) {
+    // Find objectId of the current userPreference entry
+    const objectId = (await new Parse.Query('UserPreference').equalTo('username', username).first()).toJSON().objectId
+    // Update information for that user
+    try {
+      await new Parse.Object('UserPreference')
```

Comment:
Do we really need two queries here? Asking since  I haven't looked at the documentation.

Couldn't we do something like:

```
await new Parse.Query('UserPreference').equalTo('username', username).first()
   // now set all the attributes without saving the objectId
  .set('prioritize', prioritize)
   .set(....)
```

Reply:
I just tried that way, as well as using another key like username instead of objectId so I could avoid calling that second query. However, in the first attempt of doing the .equalTo, .first and then set the values it throw me an error, whereas by passing the username as how I was passing the objectId only generates a new entry. 

I was reading documentation and it seems like there is another ways to update the information on the table. I will continue doing more research about this! 

I just created another issue to look into this topic